internal/cloud/backend_plan.go

```diff
@@ -373,6 +354,121 @@ in order to capture the filesystem context the remote workspace expects:
+func (b *Cloud) renderPlanLogs(ctx context.Context, op *backend.Operation, run *tfe.Run) error {
+	logs, err := b.client.Plans.Logs(ctx, run.Plan.ID)
+	if err != nil {
+		return err
+	}
+
+	reader := bufio.NewReaderSize(logs, 64*1024)
+
+	if b.CLI != nil {
+		for next := true; next; {
+			var l, line []byte
+			var err error
+
+			for isPrefix := true; isPrefix; {
+				l, isPrefix, err = reader.ReadLine()
+				if err != nil {
+					if err != io.EOF {
+						return generalError("Failed to read logs", err)
+					}
+					next = false
+				}
+
+				line = append(line, l...)
+			}
+
+			if next || len(line) > 0 {
+				log := &jsonformat.JSONLog{}
+				if err := json.Unmarshal(line, log); err != nil || log == nil {
+					// If we can not parse the line as JSON, we will simply
+					// print the line. This maintains backwards compatibility for
+					// users who do not wish to enable structured output in their
+					// workspace.
+					b.CLI.Output(string(line))
+					continue
+				}
+
+				// We will ignore plan output, change summary or outputs logs
+				// during the plan phase.
+				if log.Type == jsonformat.LogOutputs ||
+					log.Type == jsonformat.LogChangeSummary ||
+					log.Type == jsonformat.LogPlannedChange {
+					continue
+				}
+
+				if b.renderer != nil {
+					// Otherwise, we will print the log
+					err := b.renderer.RenderLog(log)
+					if err != nil {
+						return err
+					}
+				}
+			}
+		}
+	}
+
+	// Get the run's current status and include the workspace. We will check if
+	// the run has errored and if structured output is enabled.
+	run, err = b.client.Runs.ReadWithOptions(ctx, run.ID, &tfe.RunReadOptions{
+		Include: []tfe.RunIncludeOpt{tfe.RunWorkspace},
+	})
+	if err != nil {
+		return err
+	}
+
+	// If the run was errored, canceled, or discarded we will not resume the rest
+	// of this logic and attempt to render the plan.
+	if run.Status == tfe.RunErrored || run.Status == tfe.RunCanceled ||
+		run.Status == tfe.RunDiscarded {
+		// We won't return an error here since we need to resume the logic that
+		// follows after rendering the logs (run tasks, cost estimation, etc.)
+		return nil
+	}
+
+	// Given CLI-driven runs will silently disable SRO if the version is not supported,
+	// we need to ensure the workspace Terraform version is >=1.4.0 and has
+	// structured output enabled.
+	if satisfiesStructuredOutputTerraformVersion(run.Workspace) && b.renderer != nil {
+		token, err := b.token()
+		if err != nil {
+			return err
+		}
+		// Fetch the redacted plan.
+		redacted, err := readRedactedPlan(ctx, b.hostname, token, run.Plan.ID)
+		if err != nil {
+			return err
+		}
+
+		// Render plan output.
+		b.renderer.RenderHumanPlan(*redacted, op.PlanMode)
+	}
+
+	return nil
+}
+
+func satisfiesStructuredOutputTerraformVersion(workspace *tfe.Workspace) bool {
+	if workspace == nil {
+		return false
+	}
+
+	workspaceVersion, err := version.NewVersion(workspace.TerraformVersion)
```

Comment:
`workspace.TerraformVersion` can also be a [version constraint](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/workspaces#update-a-workspace), and not a version string, which would make this function always return false (not certain that is desirable)

This go playground illustrates the issue of trying to compare a constraint to a version. We https://go.dev/play/p/YUvyfNzsjT0

Maybe the error behavior is fine, but we should do some debug logging in this function?